src/file/imp/windows.rs

```diff
@@ -92,22 +92,22 @@ pub fn persist(old_path: &Path, new_path: &Path, overwrite: bool) -> io::Result<
     // TODO: We should probably do this in one-shot using SetFileInformationByHandle but the API is
     // really painful.
 
-    unsafe {
-        let old_path_w = to_utf16(old_path);
-        let new_path_w = to_utf16(new_path);
+    let old_path_w = to_utf16(old_path);
+    let new_path_w = to_utf16(new_path);
+
+    let flags = if overwrite {
+        MOVEFILE_REPLACE_EXISTING
+    } else {
+        0
+    };
```

Comment:
While this function is reasonably short, moving the flags away from the call site makes it a harder to read (less immediately obvious where the flags apply).

IMO, readability is significantly more important than reducing the amount of code in an unsafe block.